### PR TITLE
Junit test and code quality

### PR DESCRIPTION
**Pull request by QQH0828**
## Junit Testing
* DateTest
* DeleteScoreCommandParserTest
* JsonAdaptedScoreTest
* JsonAdaptedStudentTest
* ScoreTest
* TypicalScores

## Code quality
* DeleteScoreCommand
* UniqueScoreList
* ScoreListPanel

### CODE REVIEW

File: src/main/java/seedu/address/ui/ScoreListPanel.java [hunk 1]
```diff
@@ -147,6 +152,7 @@ private void statisticTable(Student student) {
         average.setSortable(false);
         percentage.setSortable(false);
 
+        //Due to the nature of callback and setCellFactory, it will be difficult to extract them out as a single method.
```
Comment (toh-xinyi) [inline src/main/java/seedu/address/ui/ScoreListPanel.java:155]: Can separate `//` and subsequent text with a space

File: src/test/java/seedu/address/model/score/ScoreTest.java [hunk 2]
```diff
@@ -20,4 +25,25 @@ public void constructor_null_throwsNullPointerException() {
...
+
+    @Test
+    public void isSameScore() {
+        // same object -> return true
+        assertTrue(SCORE_2.isSameScore(SCORE_2));
+
+        // null -> return false
+        assertFalse(SCORE_2.isSameScore(null));
+
+        //same label, all other attributes different -> return false
```
Comment (toh-xinyi) [inline src/test/java/seedu/address/model/score/ScoreTest.java:37]: Same comment as above

File: src/test/java/seedu/address/model/score/ScoreTest.java [hunk 3]
```diff
@@ -20,4 +25,25 @@ public void constructor_null_throwsNullPointerException() {
...
+        Score editedScore = new ScoreBuilder(SCORE_2).withDate(VALID_SCORE_DATE).withValue(VALID_SCORE_VALUE).build();
+        assertFalse(SCORE_2.isSameScore(editedScore));
+
+        //same date, all other attributes different -> return true
```
Comment (toh-xinyi) [inline src/test/java/seedu/address/model/score/ScoreTest.java:41]: ^

File: src/test/java/seedu/address/model/score/ScoreTest.java [hunk 4]
```diff
@@ -20,4 +25,25 @@ public void constructor_null_throwsNullPointerException() {
...
+        editedScore = new ScoreBuilder(SCORE_2).withValue(VALID_SCORE_VALUE).withLabel(VALID_SCORE_LABEL).build();
+        assertTrue(SCORE_2.isSameScore(editedScore));
+
+        //same value, all other attributes different -> return false
```
Comment (toh-xinyi) [inline src/test/java/seedu/address/model/score/ScoreTest.java:45]: ^

**Review by toh-xinyi — CHANGES_REQUESTED** [review]
LGTM, just a small change to be made

File: src/test/java/seedu/address/model/score/ScoreTest.java (hunk 4 above)
Comment (QQH0828) [reply src/test/java/seedu/address/model/score/ScoreTest.java:45]: Thanks for checking!

**Review by toh-xinyi — APPROVED** [review]
LGTM :)

**Pull request merged**